src/Test.UnitTests.Sarif/SarifExtensionsTests.cs

```diff
@@ -113,5 +113,29 @@ public void SarifExtensions_Result_GetMessageText_Concise()
             string actual = result.GetMessageText(rule, concise: true);
             Assert.Equal(expected, actual);
         }
+
+        [Fact]
+        public void SarifExtensions_Result_GetMessageText_Concise_Truncated()
+        {
+            var result = new Result
+            {
+                Message = new Message
+                {
+                    Id = "ruleStr1"
+                }
+            };
+
+            var rule = new ReportingDescriptor
+            {
+                MessageStrings = new Dictionary<string, MultiformatMessageString>
+                {
+                    ["ruleStr1"] = new MultiformatMessageString { Text = "First sentence is very long. Second sentence." }
+                }
+            };
+
+            const string Expected = "First sentence is ve...";
+            string actual = result.GetMessageText(rule, concise: true, maxLength: 20);
```

Comment:
compute this from your expected value (length - length of your trailing dots or ellipses.